### ISSUE TITLE
I'm getting keychain error when login

### ISSUE THREAD

**Issue by mbarciela** [EddyVerbruggen/cordova-plugin-googleplus#313]
I've been struggling with this for a long time and I'm not seeing the light at the end of the tunnel. I'm developing an mobile app with ionic and I want to add a google sign in. Like I'm using firebase I've implemented the login button with cordova-plugin-googleplus. I worked at first but recently, after a clean state and re-add the platform for iOS, it stoped working throwing the "keychain error" message on the console.

In Android it works fine; right now, I've tried everything (disable and re-enable keychain sharing capability, clean all the environment and rebuild it, check the entire GoogleService-Info.plist, checked the reversed_client_id...I don't know what I'm missing).

If someone could give me a hand I'd really appreciate it. Thanks in advanced.

Some additional info:

```
Your system information:

Cordova CLI: 6.4.0
Gulp version:  CLI version 3.9.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 1.2.1-nightly-1867
Ionic CLI Version: 2.1.4
Ionic App Lib Version: 2.1.2
ios-deploy version: 1.8.6 
ios-sim version: 5.0.11 
OS: Mac OS X Sierra
Node Version: v4.4.4
Xcode version: Xcode 8.1 Build version 8B62 
```
Plugins Installed:

```
cordova-plugin-calendar 4.5.5 "Calendar"
cordova-plugin-console 1.0.4 "Console"
cordova-plugin-device 1.1.3 "Device"
cordova-plugin-dialogs 1.3.0 "Notification"
cordova-plugin-fcm 1.1.5 "FCMPlugin"
cordova-plugin-googleplus 5.1.1 "Google SignIn"
cordova-plugin-inappbrowser 1.5.0 "InAppBrowser"
cordova-plugin-splashscreen 2.1.0 "Splashscreen"
cordova-plugin-transport-security 0.1.2 "App Transport Security"
cordova-plugin-whitelist 1.3.0 "Whitelist"
ionic-plugin-keyboard 2.2.1 "Keyboard"
```

The config.xml file (part of it):

```
  <plugin name="cordova-plugin-googleplus" spec="~5.1.1">
    <variable name="REVERSED_CLIENT_ID" value="XXXX-THIS_HAS_THE_CORRECT_ID-XXXXX"/>
  </plugin>
```
The error message on the console:
`keychain error`

Some error message from the system.log of the emulator:
`securityd[16035]: found no keychain client entitlements. task=Xxxxx App[17178]/1#0 LF=0 procid=com.xxxx.xxxx cs_flags=0x22000201`

The login function in the controller:

```
  $scope.loginGoogle = function(){
    $ionicLoading.show();
    LogService.log(TAG, 'webClientId : ' + 'XXXXX');
    var clienInfo = {};

    if(ionic.Platform.isAndroid())
    {
      clienInfo.webClientId = 'XXXXXXX';
    }

    window.plugins.googleplus.login(
      clienInfo,
      function (obj) {
        // alert(JSON.stringify(obj));

        LogService.log(TAG, 'googleplus.login result : ' + JSON.stringify(obj));
        LogService.log(TAG, 'obj.idToken : ' + obj.idToken);
        // obj.email          // 'eddyverbruggen@gmail.com'
        // obj.userId         // user id
        // obj.displayName    // 'Eddy Verbruggen'
        // obj.familyName     // 'Verbruggen'
        // obj.givenName      // 'Eddy'
        // obj.imageUrl       // 'http://link-to-my-profilepic.google.com'
        // obj.idToken        // idToken that can be exchanged to verify user identity.
        // obj.serverAuthCode // Auth code that can be exchanged for an access token and refresh token for offline access
        var token = obj.idToken;
        var userData = {
          displayName : obj.displayName,
          avatar : obj.imageUrl,
          currentSessionOn : 'google'
        };
        return AuthService.loginGoogle(token, userData).then(function (result) {
            $rootScope.$broadcast(AUTH_EVENTS.loginSuccess);
            $ionicLoading.hide();
            $state.go('app.tabs.events');
        }).catch(function(error){
            $ionicLoading.hide();
            $scope.message = error.message;
            $timeout(function() {
              $scope.message = null;
            }, 3000);
            $rootScope.$broadcast(AUTH_EVENTS.loginFailed);     
        });
      },
      function (msg) {
        LogService.logError(TAG, 'plugins.googleplus.login error: ' + msg);
        $ionicLoading.hide();
      });
  };

```

**Comment by sammuggPS** [comment]
Did you try reinstalling the plugin? Or did you just reinstall the platform?

**Comment by mbarciela** [comment]
Hi. Yes, I tried before to clean up all the environment and start from the last commit. 
After add the iOS platform again with ionic, I could see these lines:

```
Google Sign-In prerequisites

START Running hook to add iOS Keychain Sharing entitlements (required since iOS 10)

Will add iOS Keychain Sharing entitlements to project 'Grupy App'

END Running hook to add iOS Keychain Sharing entitlements (required since iOS 10)
```

**Comment by valgen** [comment]
I'm sorry I didn't see this issue and I opened a new one https://github.com/EddyVerbruggen/cordova-plugin-googleplus/issues/315 for the same problem.

Please help!

**Comment by junit38** [comment]
Hi, you must try to add this line:

<platform name="ios">
    <preference name="KeychainAccessibility" value="WhenUnlocked"/>
</platform>

in the ionic config.xml.

You can change the value as explained in the plugin's README.
(https://github.com/Crypho/cordova-plugin-secure-storage)

**Comment by mbarciela** [comment]
Didn't work for me (I got the same error). The Keychain Sharing capability is already activated as I can see in Xcode; nevertheless I tried put the line describe above but I couldn't see any difference.

**Comment by JSGuy7** [comment]
I was able to fix this error by enabling keychain sharing in xcode project manually. How I can turn keychain sharing ON automatically using cordova? I tried adding KeychainAccessibility preference to config.xml, but no luck.

**Comment by junit38** [comment]
You have to launch your project with Xcode to avoid this Error.
This works for me.
Can't fix it using ionic command line (or cordova).

**Comment by tiagoSAmaral** [comment]
@junit38  You only install and setup config.xml with this param ?

`<platform name="ios">
    <preference name="KeychainAccessibility" value="WhenUnlocked"/>
</platform>`

This not solve the problem for me. :/ i wil use Xcode for this setup.

 Waiting for news!

 Thx

**Comment by tangentlin** [comment]
I have tried @tiagoSAmaral solution, it is still having the same "found no keychain client entitlements" issue.

**Comment by Borongaj** [comment]
Same problem here, with following config :

Cordova CLI: 6.5.0
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.1.7
ios-deploy version: 1.9.0
ios-sim version: 5.0.8
OS: macOS Sierra
Node Version: v4.6.1
Xcode version: Xcode 8.2.1 Build version 8C1002

**Comment by junit38** [comment]
Hey

Have you tried to launch your project with Xcode after have enabled Keychain Sharing capability in the Capabilities section ?

This worked for me.

Fix: There is no need to install additional plugin or to add any line in config file.

**Comment by Borongaj** [comment]
Hi @junit38 ,

As you said earlier, all works fine from Xcode.

But the problem still exists from Ionic CLI.

**Comment by mbarciela** [comment]
hi, launching the project from Xcode works fine. Sorry the delay but I try to do some tests to be sure about this.

Of course, as @Borongaj said, the problem still exists from Ionic CLI.

Thanks to all.

**Comment by ashish-bhatt** [comment]
I know this issue has been closed, but in case if someone is still encountering this issue, this is how I solved it for ionic cli. I created a new oauth 2 credential using google developer console for ios, while creating it use "io.ionic.starter" as your bundle id. At the end you will get ios-url-scheme its similar to the client_id that you must have generated for android.

Goto platforms/ios/your_app/your_app-info.plist and add the following to it:
```
<dict>
        <key>CFBundleTypeRole</key>
        <string>Editor</string>
        <key>CFBundleURLName</key>
        <string>REVERSED_CLIENT_ID</string>
        <key>CFBundleURLSchemes</key>
        <array>
          <string>your_reverse_client_id_for_ios</string>
        </array>
      </dict>
```
Moreover I added the same code to the Entitlements-Debug.plist file. And you run your ionic cil app again. Hope this will help to others!
